src/CodeGen_ARM.cpp

```diff
@@ -104,17 +104,32 @@ class CodeGen_ARM : public CodeGen_Posix {
     CodeGen_ARM(const Target &);
 
 protected:
+    using codegen_func_t = std::function<Value *(int lanes, const std::vector<Value *> &)>;
     using CodeGen_Posix::visit;
 
-    /** Assuming 'inner' is a function that takes two vector arguments, define a wrapper that
-     * takes one vector argument and splits it into two to call inner. */
-    llvm::Function *define_concat_args_wrapper(llvm::Function *inner, const string &name);
+    /** Similar to llvm_type_of, but allows forcing use of Fixed of Vscale vectors. */
+    llvm::Type *llvm_type_with_constraint(const Type &t, bool scalars_are_vectors, VectorTypeConstraint constraint);
+
+    /** Define a wrapper LLVM func that takes some arguments which Halide defines
+     * and call inner LLVM intrinsic with an additional argument which LLVM requires. */
+    llvm::Function *define_intrin_wrapper(const std::string &inner_name,
+                                          const Type &ret_type,
+                                          const std::string &mangled_name,
+                                          const std::vector<Type> &arg_types,
+                                          bool add_inactive_arg,  // for SVE
```

Comment:
nit: with this many bool args, it's easy to get the order wrong at a callsite. would it be silly to consider a bitfields or struct-to-simulate-call-by-named-arg?

Reply:
Changed argument passing to just pass the flags directly.